src/Microsoft.ServiceFabric.Services/ConstantRetryPolicy.cs

```diff
@@ -0,0 +1,90 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License (MIT).See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Services.Communication.Client
+{
+    using System;
+
+    /// <summary>
+    /// Specifies the constant retry policy for retrying requests on exceptions in the communication channel between
+    /// client and service replicas.
+    /// </summary>
+    public class ConstantRetryPolicy : RetryPolicy
+    {
+        private static readonly Random Rand = new Random();
+
+        private readonly TimeSpan maxRetryBackoffIntervalOnNonTransientErrors;
+        private readonly TimeSpan maxRetryBackoffIntervalOnTransientErrors;
+        private readonly int defaultMaxRetryCount;
+        private readonly TimeSpan clientRetryTimeout;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConstantRetryPolicy"/> class with the supplied settings.
+        /// </summary>
+        /// <param name="maxRetryBackoffIntervalOnTransientErrors">
+        /// Specifies the maximum interval to back-off before retrying incase of Transient errors
+        /// </param>
+        /// <param name="maxRetryBackoffIntervalOnNonTransientErrors">
+        /// Specifies the maximum interval to back-off before retrying incase of Non transient errors
+        /// </param>
+        /// <param name="defaultMaxRetryCount">
+        /// Specifies the maximum number of times to retry.
+        /// </param>
+        /// <param name="clientRetryTimeout">Specifies the maximum time client retries the call before quitting</param>
+        public ConstantRetryPolicy(
+            TimeSpan maxRetryBackoffIntervalOnTransientErrors,
+            TimeSpan maxRetryBackoffIntervalOnNonTransientErrors,
+            int defaultMaxRetryCount,
+            TimeSpan clientRetryTimeout)
```

Comment:
Looks like both the max count and max time are mandatory. Would it be useful to specify both?

Reply:
Yes, its useful in case where Customer wants to use  this RetryCount for Transient Custom Exceptions. CLientRetryTimeout will helpful in case of non-transient errors in order to come out of the loop.